File: sub-model/src/main/java/org/github/base/entity/SysUserEntity.java
package org.github.base.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.IdType;
import org.github.base.AbstractEntity;
import java.time.LocalDate;
import java.time.LocalDateTime;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 系统-用户表
 * </p>
 *
 * @author JYD_XL
 * @since 2019-05-07
 */
@Data
@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@TableName("sys_user")
public class SysUserEntity extends AbstractEntity {

    /** UID */
    private static final long serialVersionUID = 1L;

    /** 用户ID */
    @TableId(value = "user_id", type = IdType.UUID)
    private String userId;

    /** 用户编号 */
    private String userCode;

    /** 用户名 */
    private String username;

    /** 密码 */
    private String password;

    /** 用户状态 1：启用 */
    private Boolean state;

    /** salt */
    private String salt;

    /** 是否为系统用户 */
    private Boolean isSystem;

    /** 姓名 */
    private String personName;

    /** 所属组织 */
    private String orgId;

    /** 性别 */
    private String sex;

    /** 地址 */
    private String address;

    /** 出生日期 */
    private LocalDate birthday;

    /** 学历 */
    private String edu;

    /** 用户类型，正常用户、临时用户 */
    private String userType;

    /** 手机号、电话号码 */
    private String mobile;

    /** 用户邮箱 */
    private String userEmail;

    /** 身份证号码 */
    private String idcard;

    /** 允许登录的IP区间，多个用,分割 */
    private String ipRange;

    /** 失效日期 */
    private LocalDate expiryDate;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    @TableField(fill = FieldFill.UPDATE)
    private LocalDateTime updatedAt;

    /** 删除标志 */
    @TableLogic
    private Integer deleted;

}

File: sub-model/src/main/java/org/github/base/entity/SysUserEntity.kt
package org.github.base.entity

import com.baomidou.mybatisplus.annotation.TableName
import com.baomidou.mybatisplus.annotation.IdType
import org.github.base.AbstractEntity
import com.baomidou.mybatisplus.annotation.TableId
import java.time.LocalDateTime

/**
 * <p>
 * 用户信息表
 * </p>
 *
 * @author JYD_XL
 * @since 2019-11-19
 */
@TableName("sys_user")
class SysUserEntity: AbstractEntity() {

    /** 用户ID */
    @TableId(value = "user_id", type = IdType.AUTO)
    var userId: Long? = null

    /** 部门ID */
    var deptId: Long? = null

    /** 登录账号 */
    var loginName: String? = null

    /** 用户昵称 */
    var userName: String? = null

    /** 用户类型（00系统用户） */
    var userType: String? = null

    /** 用户邮箱 */
    var email: String? = null

    /** 手机号码 */
    var phonenumber: String? = null

    /** 用户性别（0男 1女 2未知） */
    var sex: String? = null

    /** 头像路径 */
    var avatar: String? = null

    /** 密码 */
    var password: String? = null

    /** 盐加密 */
    var salt: String? = null

    /** 帐号状态（0正常 1停用） */
    var status: String? = null

    /** 删除标志（0代表存在 2代表删除） */
    var delFlag: String? = null

    /** 最后登陆IP */
    var loginIp: String? = null

    /** 最后登陆时间 */
    var loginDate: LocalDateTime? = null

    /** 创建者 */
    var createBy: String? = null

    /** 创建时间 */
    var createTime: LocalDateTime? = null

    /** 更新者 */
    var updateBy: String? = null

    /** 更新时间 */
    var updateTime: LocalDateTime? = null

    /** 备注 */
    var remark: String? = null

}
